### ISSUE TITLE
hasPermissionTo on another user model fails

### ISSUE THREAD

**Issue by spaceemotion** [spatie/laravel-permission#1011]
Hey, i got an issue with determining permissions…

I want to check if _another_ user has a specific permission, but it fails with

> There is no permission named `XXXX` for guard `web`

as I (the current user) do not have it.

It looks like it’s trying to look for the permissions in the currently authorised user, not the one I did the `$this->hasPermissionTo()` call from?

I did notice that the permissions array used internally does not contain the permission - so the error is kind of valid. The weird part is that `$this->getAllPermissions()` returns a list where my desired permission does appear in.

The call for the check is like the following:

```php
class User extends Model {
    //....

    public function isXxxxx()
    {
        return $this->hasPermissionTo('xxxx');
    }
}
````

Thanks in advance for any help!

----

**Edit:** Just checked, the following works:

```php
return $this->getAllPermissions()->where('name', 'xxxxxxxxx')->isNotEmpty();
```

**Comment by drbyte** [comment]
I think there's something in your app that's causing the unexpected result.

Here's an app I created for you which shows the `hasPermissionTo()` call working as expected:
https://github.com/drbyte/permtesting-spaceemotion1011

**Comment by spaceemotion** [comment]
@drbyte 	Sorry for the late response, and thanks for the wonderful example!

I asked around in the Laravel Slack channel and eventually resolved the issue as your example showed that it should work properly.

While I still don't know how it happened, I know that the following "fixes" I've tried did not work as intended:
- `php artisan permission:cache-reset`
  I even got the _"Permission cache flushed."_ message, but no success.
- `app(PermissionRegistrar::class)->forgetCachedPermissions();`
  calling this via code in the Migration - where I've added the permission I wanted to check against - also did not properly flush the cache.
- Setting the cache time to zero in the config file

Completely deleting the `storage/framework/cache` directory finally did the trick…
